esrally/rally.py

```diff
@@ -56,6 +56,14 @@ def runtime_jdk(v):
             except argparse.ArgumentTypeError:
                 raise argparse.ArgumentTypeError(f"must be a positive number or 'bundled' but was {v}")
 
+    def supported_es_version(v):
```

Comment:
Reading the definition of [type](https://docs.python.org/3/library/argparse.html#type) it seems to be that using type to do more advanced content based validation like here is a bit of a stretch. (hence the weird `if v:` and `return v`).

I think for now this is ok, but in a future refactoring that we've discussed about offline anyway, we could leverage [action](https://docs.python.org/3/library/argparse.html#action)?

Reply:
I was applying the same logic as for other validations but you have a valid point. I can raise a follow-up issue.

Reply:
I've raised https://github.com/elastic/rally/issues/1249 so we won't forget.